lib/common/fse_decompress.c

```diff
@@ -318,22 +318,29 @@ FORCE_INLINE_TEMPLATE size_t FSE_decompress_wksp_body(
 {
     const BYTE* const istart = (const BYTE*)cSrc;
     const BYTE* ip = istart;
-    short counting[FSE_MAX_SYMBOL_VALUE+1];
     unsigned tableLog;
     unsigned maxSymbolValue = FSE_MAX_SYMBOL_VALUE;
-    FSE_DTable* const dtable = (FSE_DTable*)workSpace;
+    short* counting = (short*) workSpace;
+    size_t const countingSize = sizeof(short) * (FSE_MAX_SYMBOL_VALUE + 1);
+    FSE_DTable* const dtable = (FSE_DTable*)(void*)((BYTE*)workSpace + countingSize);
```

Comment:
minor:
since `counting` is an array of fixed size here,
we have the opportunity to declare a `struct` describing the content of `workSpace`,
in order for pointer arithmetic to be solved automatically by the type system.